ci/test/pipeline.template.yml

```diff
@@ -162,41 +162,44 @@ steps:
           queue: linux-x86_64
         coverage: skip
 
-  - id: cargo-test
-    label: Cargo test
-    timeout_in_minutes: 30
-    artifact_paths: [junit_*.xml, target/nextest/ci/junit_cargo-test.xml]
-    inputs:
-      - Cargo.lock
-      - "**/Cargo.toml"
-      - "**/*.rs"
-      - "**/*.proto"
-      - "**/testdata/**"
-    env:
-      AWS_DEFAULT_REGION: "us-east-1"
-      # cargo-test's coverage is handled separately by cargo-llvm-cov
-      BUILDKITE_MZCOMPOSE_PLUGIN_SKIP_COVERAGE: "true"
-    plugins:
-      - ./ci/plugins/scratch-aws-access: ~
-      - ./ci/plugins/mzcompose:
-          composition: cargo-test
-    agents:
-      queue: builder-linux-x86_64
+  - group: "Cargo tests"
+    key: cargo-tests
```

Comment:
I thought you also had parallelism for cargo tests? in some other PR?

Reply:
I undid it again because I realized that most of the time of cargo tests is spent compiling, so splitting it up doesn't help much and increases cost a lot. The proper fix for this would be to build it once, then transfer the builds, but that also takes some time since Debug builds can be huge. So I didn't see a good way forward and kept the old state.